syntax = "proto2";

option go_package = "./service";

service Api {
  rpc TriggerCounter (ApiRequest) returns (CounterReply) {}
  rpc GetCounter (ApiRequest) returns (CounterReply) {}
  rpc ResetCounter (ApiRequest) returns (ApiReply) {}
}

message ApiRequest {
  optional int64 value = 1;
}

message ApiReply {
  required string message = 1;
}

message CounterReply {
  required int64 value = 1;
}